cmake/SDFUtils.cmake

```diff
@@ -52,8 +52,8 @@ endmacro (BUILD_WARNING)
 #################################################
 macro (sdf_add_library _name)
   set(LIBS_DESTINATION ${PROJECT_BINARY_DIR}/src)
-  set_source_files_properties(${ARGN} PROPERTIES COMPILE_DEFINITIONS "BUILDING_DLL")
-  add_library(${_name} SHARED ${ARGN})
+  add_library(${_name} ${ARGN})
+  set_target_properties(${_name} PROPERTIES DEFINE_SYMBOL "BUILDING_DLL")
```

Comment:
This is not directly related to this PR, but using a generic `BUILDING_DLL` macro for something that should be defined only for compilation units of sdformat can create problems if downstream projects also define the `BUILDING_DLL` macro, see https://github.com/osrf/gazebo/issues/2262 for a related Gazebo issue.

Reply:
Indeed,
replacing `BUILDING_DLL` in

https://github.com/osrf/sdformat/blob/30a669200b471b781134b170fc838b8f74e79529/include/sdf/system_util.hh#L29

with e.g. `BUILDING_SDF` would fix this.

Would it be fine to do this too in this pr?

Reply:
yeah, that would be great

Reply:
@joxoby 
Can you do this?

Reply:
Done